addon/-private/system/promise-proxies.js

```diff
@@ -82,6 +82,28 @@ export function promiseArray(promise, label) {
   });
 }
 
+export const PromiseBelongsTo = PromiseObject.extend({
+
+  // we don't proxy meta because we would need to proxy it to the relationship state container
+  //  however, meta on relationships does not trigger change notifications.
+  //  if you need relationship meta, you should do `record.belongsTo(relationshipName).meta()`
+  meta: computed(function() {
+    assert(
+      'You attempted to access meta on the promise for the async belongsTo relationship ' +
+      `${this.get('_belongsToState').internalModel.modelName}:${this.get('_belongsToState').key}'.` +
+       '\nUse `record.belongsTo(relationshipName).meta()` instead.',
+      false
+    );
+  }),
```

Comment:
sidenote: this is actually true for hasMany as well. It does proxy `meta`, however it rarely picks up the change notification (if ever).  We should address this soon.